src/lib/spinel/radio_spinel.cpp

```diff
@@ -228,18 +232,28 @@ otError RadioSpinel::CheckRadioCapabilities(otRadioCaps aRequiredRadioCaps)
 
     if ((sRadioCaps & aRequiredRadioCaps) != aRequiredRadioCaps)
     {
-        otRadioCaps missingCaps = (sRadioCaps & aRequiredRadioCaps) ^ aRequiredRadioCaps;
+        otRadioCaps missingCaps         = (sRadioCaps & aRequiredRadioCaps) ^ aRequiredRadioCaps;
+        char        strMissingCaps[200] = {0};
+        char       *ptr                 = strMissingCaps;
 
         // missingCaps may be an unused variable when LogCrit is blank
         // avoid compiler warning in that case
-        OT_UNUSED_VARIABLE(missingCaps);
-
-        LogCrit("RCP is missing required capabilities: %s%s%s%s%s",
-                (missingCaps & OT_RADIO_CAPS_ACK_TIMEOUT) ? "ack-timeout " : "",
-                (missingCaps & OT_RADIO_CAPS_TRANSMIT_RETRIES) ? "tx-retries " : "",
-                (missingCaps & OT_RADIO_CAPS_CSMA_BACKOFF) ? "CSMA-backoff " : "",
-                (missingCaps & OT_RADIO_CAPS_TRANSMIT_SEC) ? "tx-security " : "",
-                (missingCaps & OT_RADIO_CAPS_TRANSMIT_TIMING) ? "tx-timing " : "");
+        OT_UNUSED_VARIABLE(strMissingCaps);
+
+        for (uint8_t i = 1; i < static_cast<uint8_t>(sizeof(kAllRadioCapsStr)); i++)
+        {
+            if (missingCaps & (1 << i))
+            {
+                if ((ptr - strMissingCaps) + strlen(kAllRadioCapsStr[i]) >= sizeof(strMissingCaps))
+                {
+                    break;
+                }
+                sprintf(ptr, "%s ", kAllRadioCapsStr[i]);
+                ptr += strlen(kAllRadioCapsStr[i]) + 1;
+            }
+        }
+
+        LogCrit("RCP is missing required capabilities: %s", strMissingCaps);
```

Comment:
Thanks for submitting this, @Irving-cl. 👍 

How about logging the missing capabilities as a multi-line list, with each capability on its own line using a `LogCrit()` call? I think this would be simpler to implement (no need for `sprintf()`) and more scalable (if we have more capabilities in the future).